docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+for the `parse("add parallel c 20")` API call that implements the `add` command to add components.
+
+![AddSequence](diagrams/AddSequenceDiagram.png)
+
+<small><i>Figure 10</i></small>  
+
+### `CalculateCircuitCommand` <a name="calc-circ"></a>
+The calculate command can be split into two distinct sequence diagrams. Given below is the sequence diagram for the
+calculate command that does not show the access of the `component` component and shows the interactions for the
+`parse("calc power")` API call that implements this version of the `calculate` command to calculate power.  
+
+![CalcPowerSequence](diagrams/CalcPowerSequenceDiagram.png)
+
+<small><i>Figure 11</i></small>  
+
+The second sequence diagram given below shows the detailed interaction that acesses the `component` class through the
+`parse("calc reff")` API call that implements this version of the `calculate` command to calculate effective resistance.
+
+![CalcReffSequence](diagrams/CalcReffSequenceDiagram.png)
+
+<small><i>Figure 12</i></small>  
+
+--------------------------------------------------------------------------------------------------------------------
+
+## Implementation of Boolean Commands (Praveen) <a name="bool-comd"></a>
+
+![BooleanCommandClass](diagrams/BooleanCommandClassDiagram.png)
+
+<small><i>Figure 13</i></small>
+
+The second major feature of the application is the implementation of Boolean logic commands, of which various noteworthy implementation details are explained in this section. The above diagram demonstrates the relationships between the various `Command` objects. Notice the similarities to the [Circuit Commands in the earlier section](#implementation-of-circuit-commands).
+
+There are six different logic gates that can be instantiated in the program:
+- `or, and, nor, nand, xor, xnor`
+
+All gates can be first instantiated using the `Gate` class which has one `int input` and one `int output` as its attributes. 
+It has `setInput(int input)` and `getOutput()` as its methods, which are used to set the input of the logic gate and get the output of the gate respectively. 
+
+The six different logic gates take in two different inputs, which necessitates the need for a logic gate class to take in one more input.
+This can be achieved by instantiating the `TwoInputGate` class which inherits from the `Gate` class. It has an additional attribute `int secondInput`
+to take in the additional input and a method `setSecondInput(int input)` to set the Boolean value of either `0 or 1` to the additional input.
+
+The six different logic gates then individually inherit from the `TwoInputGate` class where the return value of their `getOutput()` function depends
+on the logic function of the gate itself. For instance, the `OrGate` which inherits from the `TwoInputGate` class has its `getOutput()` function set to `{return input | secondInput}`, 
+which represents the `or` operation.
```

Comment:
This is very clear and good! Maybe some visual descriptions (by way of UML comments) can be included in the below `OrGateObjectDiagram`, particularly for the "... {return input | secondInput} which represents the `or` operation" part? Also, would it be good to have a Architecture Diagram of the `Gate` classes structure? I think the names of the classes are great and intuitive, though, so this is just a suggestion (for TL;DR purposes)!

Reply:
Thanks, should be part of https://github.com/AY2021S1-CS2113T-W13-3/tp/pull/121. Updated the diagram as well: 

![OrGateObjectDiagram](https://user-images.githubusercontent.com/53231968/97584104-a098fd80-1a32-11eb-8205-1c40bb56330d.png)

I believe the diagram with the methods would then demonstrate it in its own way, as well as the Glossary definition of the OR operation:

![BooleanTemplateClassDiagram](https://user-images.githubusercontent.com/53231968/97584185-b9091800-1a32-11eb-8fbb-c55d532cb049.png)